### ISSUE TITLE
Why using multiple TXT records for the public key

### ISSUE THREAD

**Issue by HexoNerd** [Domain-Connect/spec#23]
In section 4.2.3. Security Considerations, the spec says:

> Since the public key may be greater than 255 characters, multiple TXT records may exist for the DNS TXT query.

Instead of entering multiple, enumerated TXT records, wouldn't it be better to use multiple strings inside one TXT record, like used in DKIM and SPF records?

While a single string inside a TXT record is limited to 255 chars, a singe record may consist of multiple strings.

This would save some bytes, the record for the example could look like this:

`_dcpubkeyv1 IN TXT "a=RS256,t=x509,d=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA1dCqv7JEzUOfbhWKB9mTRsv3O9Vzy1Tz3UQlIDGpnVrTPBJDQTXUhxUMREEOBKo+rOjHZqfYnSmlkgu1dnBEO8bsELQL8GjS4zsjdA53gRk2SDxuzcB4fK+NCDfnRHut5nG0S3U4cq4DuGrMDFVBwxH1duTsqDNgIOOfNTsFcWSVXoSSTqCCMGbj8Vt51umDhW" "QAj06lf5NCDfnRHut5nG0S3U4cq4DuGrMDFVBwxH1duTsqDNgIOOfNTsFcWSVXoSSTqCCMGbj8Vt51umDhWQAj06lf50qP2/jMNs2G+KTlk3dBHx3wtqYLvdcop1Tk5xBD64BPJ9uwm8KlDNHe+8O+cC9j04Ji8B2K0/PzAj90xnb8XJy/EM124hpT9lMgpHKBUvdeurJYweC6oP41gsTf5LrpjnyIy9j5FHPCQIDAQAB"`

The p= parameter would not be required anymore, the other parameters appear only once.

Concatenation as described in RFC7208 (SPF) section 3.3:
https://tools.ietf.org/html/rfc7208#section-3.3

**Comment by pawel-kow** [comment]
@arnoldblinn, can you take a look? This is the part you drafted this way.

**Comment by arnoldblinn** [comment]
I have seen some UX for TXT records limit you to 255, and may have lept to assumption that this was a limitation in the protocol.  As such, there is no reason to put it into one record. But should it need to be split, it can be.  I'll add that we should test this with more providers and see the behavior.

I'll see about getting the key for exampleservice.domainconnect.org put into one record so we can try it out....

**Comment by alinconstantin** [comment]
Arnold, you are correct.
There is no limitation of the DNS protocol, and for those that can set the value in one record there should be no requirement to set p=1, part.
However, there are UI limitations in setting the value - when I set the value for office.com I tried setting in one value, and the scripts used to set values in MSFT DNS didn't like it, so I had to set it in two TXT records p=1 and p=2.
Please keep that part in spec if you relax the requirement for those who can store in one record.

Thanks,
Alin
________________________________

**Comment by arnoldblinn** [comment]
Whle I would like to claim I anticipated this, I'm more lucky than smart here.

There are two sides of the public key fetch.  One is the producer of the data. The other the consumer.

You've verified for me there are some producers that can't deal with longer strings.  I've gone in and verified that at least for my consumer (the example service) that I can in fact deal with longer strings.

I've modified the specification to read appropriately with my latest pull request.
